pkg/tfshim/shim.go

```diff
@@ -48,6 +48,7 @@ type InstanceDiff interface {
 	ProposedState(res Resource, priorState InstanceState) (InstanceState, error)
 	Destroy() bool
 	RequiresNew() bool
+	DiffEqualDecisionOverride() *bool
```

Comment:
Trying to understand this, how is this different from HasNoChanges()? Why would you need both in the model, can you give an example?

Reply:
Yeah, `DiffEqualDecisionOverride` is like a feature flag - shim implementations which return a non-null value signify that this decision needs to be respected.

`HasNoChanges` seems to only be used in tests - we could rename `DiffEqualDecisionOverride` to something else and use `HasNoChanges` for the result. Or we could clean up `HasNoChanges`

Reply:
> `HasNoChanges` seems to only be used in tests

I don't think this is true. I have seen diff status corrections from `HasNoChanges` in the wild.

Reply:
My bad, I deleted its only usage and then could not find it being used outside of tests...

Reply:
Do you have concrete examples of this @iwahbe? The only ones I could find which were in GCP labels and AWS tags.

Reply:
```
pkg/tfshim/shim.go
49:	HasNoChanges() bool
pkg/tfbridge/provider.go
1158:	if !diff.HasNoChanges() {
pkg/tfshim/tfplugin5/provider_test.go
1156:			if diff.HasNoChanges() {

```

It's used. IDE to the rescue.

Reply:
They are implemented differently - `HasNoChanges` used the TF Diff which we are trying to get rid of. I think for now we have to keep both around